### DISCUSSION TITLE
Graphannis crashing

### DISCUSSION THREAD

**Discussion by gcelano** [korpling/ANNIS#862] [General]
Annis server (4.11 and 4.12) works, but keeps repeating the error you can read in the attached screeshot. What can be the cause of it?
![Screenshot from 2024-07-03 12-22-21](https://github.com/korpling/ANNIS/assets/5678472/fac49963-9094-44bb-aeea-6779f1e00c54)

**Comment by thomaskrause** [comment]
Based on the error message it seems that the previous ANNIS service did not shut down correctly or was somehow aborted. There is still a `graphannis-webservice` process running, that is locking the database directory. You can search for this process e.g.  in `top` and kill it, or restart the server (which would stop any proccesses).

**Reply by gcelano** [reply]
Yes: sometimes, graphannis keeps running even after stopping Annis, therefore one should always check that graphannis is not working after stopping Annis.

**Reply by thomaskrause** [reply]
The service configuration should gently kill the service when running `systemctl stop annis.service` and also stop the graphANNIS webservice process. This might be related to #861 and a problem with the configured user that is supposed to execute the service. If the `annis` user would not exist and the process is started anyway, but the `celano` user, stopping the service might fail as well. Can you adjust the configured user that should start the service and check that the service is actually stopped with `systemctl stop annis.service`?

**Reply by gcelano** [reply]
`systemctl stop annis.service` stops `annis` and `graphannis` correctly.
